include/gazebo_motor_failure_plugin.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 Nuno Marques, PX4 Pro Dev Team, Lisbon
+ * Copyright 2017 Siddharth Patel, NTU Singapore
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <boost/bind.hpp>
+#include <gazebo/gazebo.hh>
+#include <gazebo/common/common.hh>
+#include <gazebo/common/Plugin.hh>
+#include "gazebo/transport/transport.hh"
+#include "gazebo/msgs/msgs.hh"
+#include "common.h"
+//#include "int.pb.h"
+
+// ROS Topic subscriber
+#include <thread>
+#include "ros/ros.h"
+#include "ros/callback_queue.h"
+#include "ros/subscribe_options.h"
+#include <std_msgs/Int32.h>
+
+namespace gazebo {
+
+// Default values
+static const std::string kDefaultROSMotorNumSubTopic = "/motor_failure/motor_number";
+static const std::string kDefaultMotorFailureNumPubTopic = "/gazebo/motor_failure_num";
+
+static constexpr int kDefaultMotorFailureNum = 0;
+
+class GazeboMotorFailure : public ModelPlugin {
+ public:
+
+  void motorFailNumCallBack(const std_msgs::Int32ConstPtr& _msg) {
+    this->motor_Failure_Number_ = _msg->data;
+    //std::cout << "[gazebo_motor_failure_plugin]: Subscribe to /motor_failure/motor_number " << std::endl;
```

Comment:
These can be kept if we want to debug further.

Reply:
noted